pkg/infrastructure/nutanix/clusterapi/clusterapi.go

```diff
@@ -0,0 +1,212 @@
+		logrus.Infof("waiting the image data uploading from %s, taskUUID: %s.", imgURI, taskUUID)
+
+		// Wait till the image creation task is successed.
+		if err = nutanixtypes.WaitForTask(nutanixCl.V3, taskUUID); err != nil {
+			e1 := fmt.Errorf("failed to create the rhcos image %q. %w", imgName, err)
+			logrus.Error(e1)
+			return e1
+		}
+		logrus.Infof("Created and uploaded the rhcos image data %s (uuid: %s).", imgName, imgUUID)
+	} else {
+		err = fmt.Errorf("failed to convert the task UUID %v to string", respi.Status.ExecutionContext.TaskUUID)
+		logrus.Errorf(err.Error())
+		return err
+	}
+
+	return nil
+}
+
+// Ignition handles preconditions for bootstrap ignition and
+// generates ignition data for the CAPI bootstrap ignition secret.
+// Load the ignition iso image to prism_central.
+func (p Provider) Ignition(ctx context.Context, in infracapi.IgnitionInput) ([]byte, error) {
+	ic := in.InstallConfig.Config
+	nutanixCl, err := nutanixtypes.CreateNutanixClientFromPlatform(ic.Platform.Nutanix)
+	if err != nil {
+		return nil, fmt.Errorf("fail to create nutanix client. %w", err)
+	}
+
+	imgName := nutanixtypes.BootISOImageName(in.InfraID)
+	imgPath, err := nutanixtypes.CreateBootstrapISO(in.InfraID, string(in.BootstrapIgnData))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create bootstrap ignition iso. %w", err)
+	}
+
+	// upload the bootstrap image.
+	imgReq := &nutanixclientv3.ImageIntentInput{}
+	imgSpec := &nutanixclientv3.Image{
+		Name:        &imgName,
+		Description: ptr.To("Created By OpenShift Installer"),
+		Resources: &nutanixclientv3.ImageResources{
+			ImageType: ptr.To("ISO_IMAGE"),
+		},
+	}
+	imgReq.Spec = imgSpec
+	categoryKey := nutanixtypes.CategoryKey(in.InfraID)
+	imgMeta := &nutanixclientv3.Metadata{
+		Kind:       ptr.To("image"),
+		Categories: map[string]string{categoryKey: nutanixtypes.CategoryValueOwned},
+	}
+	imgReq.Metadata = imgMeta
+
+	// create the image object.
+	respi, err := nutanixCl.V3.CreateImage(ctx, imgReq)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create the bootstrap image %q. %w", imgName, err)
+	}
+	imgUUID := *respi.Metadata.UUID
+
+	if taskUUID, ok := respi.Status.ExecutionContext.TaskUUID.(string); ok {
+		logrus.Infof("Creating the bootstrap image %s (uuid: %s), taskUUID: %s.", imgName, imgUUID, taskUUID)
+
+		// Wait till the image creation task is successed.
+		if err = nutanixtypes.WaitForTask(nutanixCl.V3, taskUUID); err != nil {
+			err = fmt.Errorf("failed to create the bootstrap image %q. %w", imgName, err)
+			logrus.Errorf(err.Error())
+			return nil, err
+		}
+		logrus.Infof("Created the bootstrap image %s (uuid: %s).", imgName, imgUUID)
+	} else {
+		err = fmt.Errorf("failed to convert the task UUID %v to string", respi.Status.ExecutionContext.TaskUUID)
+		logrus.Error(err)
+		return nil, err
+	}
+
+	// upload the image data.
+	logrus.Infof("To upload the bootstrap image %s (uuid: %s) data from file %s", imgName, imgUUID, imgPath)
```

Comment:
I don't understand this log message. Maybe it is something like "Preparing to upload the bootstrap image"?